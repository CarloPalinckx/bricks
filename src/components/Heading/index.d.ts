import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import { StyledType } from '../../utility/styled';
declare type HeadingHierarchyThemeType = {
    color: string;
    fontFamily: string;
    fontSize: string;
    fontWeight: string;
    lineHeight: string;
    textTransform: string;
};
declare type HeadingThemeType = StyledType & {
    default: {
        color: string;
        fontFamily: string;
        fontSize: string;
        fontWeight: string;
        lineHeight: string;
        textTransform: string;
    };
    hierarchy: {
        hierarchy1: HeadingHierarchyThemeType;
        hierarchy2: HeadingHierarchyThemeType;
        hierarchy3: HeadingHierarchyThemeType;
        hierarchy4: HeadingHierarchyThemeType;
        hierarchy5: HeadingHierarchyThemeType;
        hierarchy6: HeadingHierarchyThemeType;
        [key: string]: HeadingHierarchyThemeType;
    };
};
declare type PropsType = StyledType & {
    hierarchy?: 1 | 2 | 3 | 4 | 5 | 6;
    element?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'span' | 'p';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
};
declare const StyledHeading: _S<PropsType, _T, Pick<PropsType, "className" | "innerRef" | "hierarchy" | "element" | "textAlign"> & {
    theme?: _T | undefined;
}>;
export default StyledHeading;
export { PropsType, HeadingThemeType, HeadingHierarchyThemeType };
