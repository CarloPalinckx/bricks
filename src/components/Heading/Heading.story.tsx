import { select } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Heading from '../Heading';

storiesOf('Heading', module)
    .add('default', () => (
        <Heading>
            This is a div element.
        </Heading>
    ))
    .add('With a hierarchy', () => {
        const hierarchy = select(
            'Hierarchy',
            [1, 2, 3, 4, 5, 6],
            1
        );

        return (
            <Heading
                hierarchy={hierarchy}
            >
                This is an h{hierarchy} element.
            </Heading>
        );
    });
