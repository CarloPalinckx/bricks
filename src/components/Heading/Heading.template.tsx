import React, { StatelessComponent } from 'react';
import { StyledType } from '../../utility/styled';

type PropsType = StyledType & {
    hierarchy?:
        |1
        |2
        |3
        |4
        |5
        |6;
};

const Heading:StatelessComponent<PropsType> = (props):JSX.Element => {
    const Element = props.hierarchy !== undefined
        ? `h${props.hierarchy}`
        : 'div';

    return (
        <Element role="heading" className={props.className}>
            {props.children}
        </Element>
    );
};

export default Heading;
export {
    PropsType,
};
