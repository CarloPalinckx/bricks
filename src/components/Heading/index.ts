export { default, HeadingThemeType } from './Heading.style';
