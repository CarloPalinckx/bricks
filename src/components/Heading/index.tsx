import React, { SFC } from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import styled, { StyledType } from '../../utility/_styled';

type HeadingHierarchyThemeType = {
    fontFamily: string;
    fontSize: string;
    fontWeight: {
        light: string;
        default: string;
    };
    lineHeight: string;
    color: string;
};

type HeadingThemeType = StyledType & {
    hierarchy1: HeadingHierarchyThemeType;
    hierarchy2: HeadingHierarchyThemeType;
    hierarchy3: HeadingHierarchyThemeType;
    hierarchy4: HeadingHierarchyThemeType;
    hierarchy5: HeadingHierarchyThemeType;
    hierarchy6: HeadingHierarchyThemeType;
    [key: string]: HeadingHierarchyThemeType;
};

type PropsType = StyledType & {
    hierarchy?: 1 | 2 | 3 | 4 | 5 | 6;
    element?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'span' | 'p';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    light?: boolean;
};

const HeadingElement: SFC<PropsType> = (props): JSX.Element => {
    const Element = props.element !== undefined ? `${props.element}` : 'div';

    return (
        <Element role="heading" className={props.className}>
            {props.children}
        </Element>
    );
};

const StyledHeading = styled(HeadingElement)`
    color: ${({ hierarchy, theme }): string =>
        !hierarchy ? theme.Heading.hierarchy1.color : theme.Heading[`hierarchy${hierarchy}`].color};
    font-family: ${({ hierarchy, theme }): string =>
        !hierarchy ? theme.Heading.hierarchy1.fontFamily : theme.Heading[`hierarchy${hierarchy}`].fontFamily};
    font-size: ${({ hierarchy, theme }): string =>
        !hierarchy ? theme.Heading.hierarchy1.fontSize : theme.Heading[`hierarchy${hierarchy}`].fontSize};
    font-weight: ${({ light, hierarchy, theme }): string => {
        if (light && hierarchy) {
            return theme.Heading[`hierarchy${hierarchy}`].fontWeight.light;
        } else if (!light && hierarchy) {
            return theme.Heading[`hierarchy${hierarchy}`].fontWeight.default;
        } else if (light && !hierarchy) {
            return theme.Heading.hierarchy1.fontWeight.light;
        }

        return theme.Heading.hierarchy1.fontWeight.default;
    }};
    line-height: ${({ hierarchy, theme }): string =>
        !hierarchy ? theme.Heading.hierarchy1.lineHeight : theme.Heading[`hierarchy${hierarchy}`].lineHeight};

    text-align: ${({ textAlign }): string => (textAlign !== undefined ? textAlign : '')};
    margin: 0;
    -webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
`;

export default StyledHeading;
export { PropsType, HeadingThemeType, HeadingHierarchyThemeType };
