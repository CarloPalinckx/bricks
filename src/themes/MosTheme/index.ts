export { default } from './MosTheme.template';
export { default as mosTheme } from './MosTheme.theme';
