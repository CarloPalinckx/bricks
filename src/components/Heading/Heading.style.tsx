import { StyledComponentClass as _S } from 'styled-components';
import styled from '../../utility/styled';
import Heading from './Heading.template';

type HeadingThemeType = {
    color:string;
    fontFamily:string;
    fontSize:string;
    fontWeight:string;
};

const StyledHeading = styled(Heading)`
    color: ${({ theme }):string => theme.Heading.color};
    font-family: ${({ theme }):string => theme.Heading.fontFamily};
    font-size: ${({ theme }):string => theme.Heading.fontSize};
    font-weight: ${({ theme }):string => theme.Heading.fontWeight};
    line-height: 1.5;
    margin: 0;
`;

export default StyledHeading;
export {
    HeadingThemeType,
};
