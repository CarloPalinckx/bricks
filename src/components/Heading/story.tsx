import { select, boolean } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Heading, { PropsType } from '.';

storiesOf('Heading', module)
    .add('Default', () => {
        const hierarchy = select('Hierarchy', [1, 2, 3, 4, 5, 6] , 1) as PropsType['hierarchy'];
        const element = select('Element', ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'div', 'span', 'p'] , 'h1') as PropsType['element'];
        const textAlign = select('Text-align', ['left', 'right', 'center', 'justify'], 'left') as PropsType['textAlign'];
        const light = boolean('light', false) as PropsType['light'];

        return (
            <Heading hierarchy={hierarchy} element={element} textAlign={textAlign} light={light}>
                This is an {element !== undefined ? `${element}` : 'div'} element, with {hierarchy !== undefined ? `h${hierarchy}` : 'h1'} styling.
            </Heading>
        );
    });
