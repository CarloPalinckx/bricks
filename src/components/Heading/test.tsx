import toJson from 'enzyme-to-json';
import React from 'react';
import Heading from '.';
import { shallowWithTheme } from '../../utility/styled/testing';
import MosTheme from '../../themes/MosTheme/MosTheme.theme';

describe('Heading', () => {
    it('should render a Heading component with default styling', () => {
        const defaultHeading = shallowWithTheme(<Heading />);

        expect(toJson(defaultHeading.dive())).toMatchSnapshot();
    });

    it('should render a Heading component with different hierarchy styles', () => {
        const h1 = shallowWithTheme(<Heading hierarchy={1} />);
        const h2 = shallowWithTheme(<Heading hierarchy={2} />);
        const h3 = shallowWithTheme(<Heading hierarchy={3} />);
        const h4 = shallowWithTheme(<Heading hierarchy={4} />);
        const h5 = shallowWithTheme(<Heading hierarchy={5} />);
        const h6 = shallowWithTheme(<Heading hierarchy={6} />);

        /* tslint:disable */
        (expect(toJson(h1.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy1.fontSize);
        (expect(toJson(h2.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy2.fontSize);
        (expect(toJson(h3.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy3.fontSize);
        (expect(toJson(h4.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy4.fontSize);
        (expect(toJson(h5.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy5.fontSize);
        (expect(toJson(h6.dive())) as any).toHaveStyleRule('font-size', MosTheme.Heading.hierarchy6.fontSize);
        /* tslint:enable */
    });

    it('should render a Heading component with different elements', () => {
        const divHeading = shallowWithTheme(<Heading element="div" />);
        const spanHeading = shallowWithTheme(<Heading element="span" />);
        const pHeading = shallowWithTheme(<Heading element="p" />);
        const h1Heading = shallowWithTheme(<Heading element="h1" />);
        const h2Heading = shallowWithTheme(<Heading element="h2" />);
        const h3Heading = shallowWithTheme(<Heading element="h3" />);
        const h4Heading = shallowWithTheme(<Heading element="h4" />);
        const h5Heading = shallowWithTheme(<Heading element="h5" />);
        const h6Heading = shallowWithTheme(<Heading element="h6" />);

        /* tslint:disable */
        expect(toJson(divHeading.dive()).type).toBe('div');
        expect(toJson(spanHeading.dive()).type).toBe('span');
        expect(toJson(pHeading.dive()).type).toBe('p');
        expect(toJson(h1Heading.dive()).type).toBe('h1');
        expect(toJson(h2Heading.dive()).type).toBe('h2');
        expect(toJson(h3Heading.dive()).type).toBe('h3');
        expect(toJson(h4Heading.dive()).type).toBe('h4');
        expect(toJson(h5Heading.dive()).type).toBe('h5');
        expect(toJson(h6Heading.dive()).type).toBe('h6');
        /* tslint:enable */
    });

    it('should render a Heading component with light fontWeight', () => {
        const h1Light = shallowWithTheme(<Heading hierarchy={1} light={true} />);
        const h2Light = shallowWithTheme(<Heading hierarchy={2} light={true} />);
        const h3Light = shallowWithTheme(<Heading hierarchy={3} light={true} />);
        const h4Light = shallowWithTheme(<Heading hierarchy={4} light={true} />);
        const h5Light = shallowWithTheme(<Heading hierarchy={5} light={true} />);
        const h6Light = shallowWithTheme(<Heading hierarchy={6} light={true} />);
    const defaultLight = shallowWithTheme(<Heading light={true} />);

        /* tslint:disable */
        (expect(toJson(h1Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy1.fontWeight.light);
        (expect(toJson(h2Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy2.fontWeight.light);
        (expect(toJson(h3Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy3.fontWeight.light);
        (expect(toJson(h4Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy4.fontWeight.light);
        (expect(toJson(h5Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy5.fontWeight.light);
        (expect(toJson(h6Light.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy6.fontWeight.light);
        (expect(toJson(defaultLight.dive())) as any).toHaveStyleRule('font-weight', MosTheme.Heading.hierarchy1.fontWeight.light);
        /* tslint:enable */
    });
});
