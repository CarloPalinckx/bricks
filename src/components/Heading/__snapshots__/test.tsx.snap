// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a Heading component with default styling 1`] = `
.c0 {
  color: #333740;
  font-family: Melbourne,sans-serif;
  font-size: 36px;
  font-weight: 700;
  line-height: 45px;
  margin: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<div
  className="c0"
  role="heading"
/>
`;
